### PR TITLE
Fix #12193.

### LINKED ISSUE #12193: add writemime(::IO, ::MIME"text/markdown", x::Markdown.MD) method
The latest IJulia master version is capable of sending "text/markdown" output directly to the front-end (which supports displaying it directly).  However, somewhat perversely, the `Markdown.MD` type does not support rendering (via `writemime`) to text/markdown.

(More generally, it would be nice to be able to get Markdown output back from markdown inputs.)

cc: @one-more-minute

### PR DESCRIPTION
**Pull request by MichaelHatherly**
cc @stevengj, ref https://github.com/JuliaLang/julia/issues/12193

### CODE REVIEW

File: base/markdown/render/plain.jl [hunk 1]
```diff
@@ -83,3 +83,4 @@ plaininline(io::IO, x) = writemime(io, MIME"text/plain"(), x)
 # writemime
 
 Base.writemime(io::IO, ::MIME"text/plain", md::MD) = plain(io, md)
+Base.writemime(io::IO, ::MIME"text/markdown", md::MD) = plain(io, md)
```
Comment (stevengj) [inline base/markdown/render/plain.jl]: Maybe just:
```
Base.writemime(io::IO, ::Union(MIME"text/plain",MIME"text/markdown"), md::MD) = plain(io, md)
```
?

Comment (MichaelHatherly) [inline base/markdown/render/plain.jl]: Oh, yeah... that makes more sense. With ``Union{...}`` rather than ``Union(...)`` though?

Comment (MichaelHatherly) [inline base/markdown/render/plain.jl]: Doing that actually gives a ambiguity warning though:

```
markdown/Markdown.jl
WARNING: New definition 
    writemime(Base.IO, Union{Base.Multimedia.MIME{:text/plain}, Base.Multimedia.MIME{:text/markdown}}, Base.Markdown.MD) at markdown/render/plain.jl:86
is ambiguous with: 
    writemime(Base.IO, Base.Multimedia.MIME{:text/plain}, Any) at replutil.jl:4.
To fix, define 
    writemime(Base.IO, Base.Multimedia.MIME{:text/plain}, Base.Markdown.MD)
before the new definition.
```

Comment (stevengj) [inline base/markdown/render/plain.jl]: Oh well.

**Comment by MichaelHatherly** [comment]
Thanks.

**Comment by tkelman** [comment]
Are there tests for `writemime`?

**Comment by MichaelHatherly** [comment]
There's tests for all the methods (``plain``, ``html``, ``latex``) that the ``writemime`` methods call. Nothing that tests the entire code path though. I'll add some later today/tomorrow if I've got time.

**Pull request merged**